### PR TITLE
Check for recently placed identical orders during balancing

### PR DESCRIPTION
**Pull request by nicolaslwilson**
Creates a `Map` that stores recently placed orders along with a timestamp. During balancing we check if an identical order has been placed recently before placing the order. After placing an order we record the order in the Map.

### CODE REVIEW

File: src/liquidator.ts [hunk 1]
```diff
@@ -1157,6 +1191,27 @@ async function closePositions(
...
+) {
+  return side + size + type + marketIndex;
+}
+
+function isIdenticalOrderRecentlyPlaced(key: OrderMapKey) {
+  const now = Date.now();
+  return now - ordersPlaced.get(key) < orderDupeSafteyWindow;
+}
+
+function updateOrdersPlacedMap(orderMapKey: string) {
```
Comment (nicolaslwilson) [inline src/liquidator.ts:1211]:
```suggestion
function updateOrdersPlacedMap(orderMapKey: OrderMapKey) {
```

File: src/liquidator.ts [hunk 2]
```diff
@@ -1157,6 +1191,27 @@ async function closePositions(
...
+  size: number,
+  type: 'perp' | 'spot',
+  marketIndex: number,
+) {
+  return side + size + type + marketIndex;
+}
+
+function isIdenticalOrderRecentlyPlaced(key: OrderMapKey) {
+  const now = Date.now();
+  return now - ordersPlaced.get(key) < orderDupeSafteyWindow;
```
Comment (riordanp) [inline src/liquidator.ts:1208]: `ordersPlaces.get(key)` is possibly undefined, should return false in this case I guess

Comment (nicolaslwilson) [reply src/liquidator.ts:1208]: Ah yeah, `now - undefined` evaluates to `NaN` and so the comparison will evaluate to false in that case but that's more of a JavaScript-y way to handle it. I'll update it to make `tsc` happy 👍

**Comment by nicolaslwilson** [comment]
Fixed by setting the `reduceOnly` flag to true in https://github.com/blockworks-foundation/liquidator-v3/commit/8be33e400c7e702b4b9f85f7fa6acb6e067a2bcc so closing this.

**Pull request closed without merging**